### PR TITLE
WIP: Calendar Selection Option

### PR DESCRIPTION
**Pull request by KarolinaXin**
Still working on it

### CODE REVIEW

**Review by gregmaxin — CHANGES_REQUESTED** [review]
- The package.json file needs to be updated. Use npm install --save {name} to have it saved to the package.json automatically

File: src/pages/Search/Search.js [hunk 1]
```diff
@@ -85,6 +105,9 @@ export default class Search extends Component {
       }   
       this.setState({locations: locations});
       this.setState({location: locations[0]});
+      this.setState({selectedDates: []});
```
Comment (Ruxuan) [inline src/pages/Search/Search.js:108]:
```
this.setState({
  locations: locations,
  location: locations[0],
  selectedDates: [],
  from: undefined,
  to: undefined
})
```

File: src/pages/Search/Search.js [hunk 2]
```diff
@@ -140,6 +163,53 @@ export default class Search extends Component {
           });
       }
   }
+  handleDateOptionDisplay = (selectedDateValue) => {
+    console.log("YOLO");
```
Comment (Ruxuan) [inline src/pages/Search/Search.js:167]: Remove `console.log` lines

File: src/pages/Search/Search.js [hunk 3]
```diff
@@ -279,18 +395,21 @@ export default class Search extends Component {
...
                   <Select className="dropdown"
-                    value={daysOfWeek}
-                    onChange={this.handleChange8}
-                    options={daysOfWeekOptions}
+                    value={selectDates}
+                    onChange={this.handleDateOptionSelection}
+                    options={selectDatesOptions}
                   />
+                  </div>
+                  {this.handleDateOptionDisplay(selectDates.value)}
```
Comment (Ruxuan) [inline src/pages/Search/Search.js:412]: The default value of `selectDates.value` is `dayOfWeek` which causes a popup. So when a user first navigates to this page, the popup is shown.
Screenshot 2019-07-15 14 32 08

File: src/pages/Search/Search.js [hunk 4]
```diff
@@ -140,6 +163,53 @@ export default class Search extends Component {
...
+    console.log(typeof selectedDateValue);
+    const {from, to} = this.state;
+    const modifiers = {start: from, end: to};
+    if (selectedDateValue === undefined) return;
+    console.log('dateOption Choice ' + selectedDateValue);
+    //trigger={<button> Trigger</button>}
+    if (selectedDateValue === 'dayOfWeek') {
+        return (
+            <>
+              <div className="headingRow"> Day of the Week </div>
```
Comment (Ruxuan) [inline src/pages/Search/Search.js:178]: I think the "Day of the week" div should be inside of `<Popup>` then you can remove the `<>`
Screenshot 2019-07-15 14 38 30

File: src/pages/Search/Search.js [hunk 5]
```diff
@@ -140,6 +163,53 @@ export default class Search extends Component {
...
+                    onChange={this.handleChange8}
+                    options={daysOfWeekOptions}
+                  />
+              </Popup>
+
+            </>
+        );
+      } else if (selectedDateValue === 'individualDates') {
+        return (
+              <Popup open = {selectedDateValue === 'individualDates'} position="right center">
```
Comment (Ruxuan) [inline src/pages/Search/Search.js:191]: I'm a bit confused here. Is there a reason why `individualDates` pops up but `dateRange` doesn't?

**Review by Ruxuan — COMMENTED** [review]
Need to add `react-calendar-multiday`, `react-day-picker`, `reactjs-popup` to the package.json

**Pull request closed without merging**